contrib/automation_tests/test_cases/symbols_tab.py

```diff
@@ -153,3 +154,100 @@ def _try_verify_functions_are_hooked(self):
+    the path to hello_ggp's main.c source code file. This path is hard-coded to match the
+    E2E workflow in testrunner, but can be overwritten by the environment variable
+    `ORBIT_SOURCE_CODE_VIEW_E2E_TEST_FILE_PATH`.
+
+    Selection is done be filtering the function list and loading the first remaining row.
+    """
+
+    def _provoke_goto_source_action(self, function_search_string: str):
+        _show_symbols_and_functions_tabs(self.suite.top_window())
+
+        logging.info('Start showing source code for function {}'.format(
+            function_search_string))
+        functions_dataview = DataViewPanel(
+            self.find_control("Group", "FunctionsDataView"))
+
+        logging.info('Waiting for function list to be populated...')
+        wait_for_condition(lambda: functions_dataview.get_row_count() > 0, 100)
+
+        logging.info('Filtering functions')
+        functions_dataview.filter.set_focus()
+        functions_dataview.filter.set_edit_text('')
+        send_keys(function_search_string)
+        wait_for_condition(lambda: functions_dataview.get_row_count() == 1)
+        functions_dataview.get_item_at(0, 0).click_input('right')
+
+        logging.info('Click on "Go to Source code"')
+        self.find_context_menu_item('Go to Source code').click_input()
+
+    def _handle_file_not_found_message_box(self):
+        logging.info('Waiting for message box')
+
+        def get_message_box():
+            return self.find_control(control_type="Window", name="Source code file not found", parent=self.suite.top_window(), recurse=False)
+        wait_for_condition(lambda: get_message_box().is_visible())
+
+        logging.info('Message box found - Clicking "Choose file..."')
+        choose_file_button = self.find_control(
+            control_type="Button", name="Choose file...", parent=get_message_box(), recurse=True)
+        choose_file_button.click_input()
+
+    def _handle_file_open_dialog(self):
+        logging.info("Waiting for File Open Dialog")
+
+        def get_file_open_dialog():
+            return self.find_control(control_type="Window", name_contains="Choose ", parent=self.suite.top_window(), recurse=False)
+        wait_for_condition(lambda: get_file_open_dialog().is_visible())
+        file_open_dialog = get_file_open_dialog()
+        logging.info(
+            "File Open Dialog is now visible. Looking for file edit...")
+
+        logging.info("File Edit was found. Entering file path...")
+        file_edit = self.find_control(
+            control_type="Edit", name="File name:", parent=file_open_dialog, recurse=True)
+        file_edit.set_focus()
+        file_edit.set_edit_text('')
+
+        # The environment variable allows to overwrite the path easily for local testing
+        file_path = os.getenv(
+            "ORBIT_SOURCE_CODE_VIEW_E2E_TEST_FILE_PATH", "C:\\build\\scratch\\test\\hello_ggp\\src\\main.c")
```

Comment:
Could you use a relative path instead and keep the file next to the test itself so you wouldn't have to configure the test, or is that problematic due to open source? This is quite opaque to devs who want to run the test locally - wouldn't a command line parameter (using absl flags) a better option if a relative path can't be used?

How does the file get to this location on the testing machines? It's not mentioned in the comments for the test.

Reply:
Yes, several points here:

1. Yes, we can't check-in this file into our repo since it's not open source, which means we need to download it from an internal bucket to the E2E test machine. And even if we could, this would mean we need to package this random C-file with the official distribution of Orbit which sounds not ideal to me.
2. Yes a command line parameter does work as well. I will add it.
3. I will improve the documentation.

Reply:
I'd argue that the random c-file is test data and should, as such, be distributed in a data-directory with the tests. But anyways, doesn't matter since we can't do that. In that case the configurable path solution sounds good.

Reply:
We could go through the process of open sourcing hello_ggp, but it might not be worth the effort. But wanted to mention that the option is there.

Reply:
Yes, good point. I also thought about this and concluded that it's not worth it. I assume the whole process will take several weeks with at least a dev-week of work.

Reply:
Agree - we may want to keep this in mind if we end up using more of the source code across tests, but I can't see that happening, I don't think it's worth it for a single test.